golem-common/src/model/mod.rs

```diff
@@ -520,6 +521,21 @@ impl WorkerMetadata {
     }
 }
 
+impl From<WorkerMetadata> for golem_api_grpc::proto::golem::worker::WorkerMetadata {
+    fn from(value: WorkerMetadata) -> Self {
+        golem_api_grpc::proto::golem::worker::WorkerMetadata {
+            worker_id: Some(value.worker_id.worker_id.into_proto()),
+            account_id: Some(value.account_id.into()),
+            args: value.args,
+            env: HashMap::from_iter(value.env.iter().cloned()),
+            template_version: value.worker_id.template_version,
+            status: Into::<golem::worker::WorkerStatus>::into(value.last_known_status.status)
+                .into(),
+            retry_count: 0, // FIXME
```

Comment:
Maybe we should not have this here because we will forget that it cannot set the retry_count and misuse it 

Right now the get-metadata gRPC endpoint in worker executor is like this: https://github.com/golemcloud/golem/blob/main/golem-worker-executor-base/src/grpc.rs#L651-L669

But that call to calculate the retry count requires reading from the oplog, so I think it is good in general if worker enumeration returns 0, except if the `precise` flag is set.

I think in longer term the retry count should also be cached in the `last_known_status` but that change should not be part of this PR.

So for now I would remove this from here and have two more custom conversion functions somewhere in `worker-executor-base` that is used both by the worker enumeration, and the get-metadata endpoint - one that uses the last known status and sets retry count to 0 (for now), and one that calls the necessary functions to calculate the latest information ("precise mode").